engine/launcher/src/main/scala/org/enso/launcher/FileSystem.scala

```diff
@@ -0,0 +1,55 @@
+package org.enso.launcher
+
+import java.io.PrintWriter
+
+import org.apache.commons.io.FileUtils
+import java.nio.file.{Files, Path}
+
+import org.enso.launcher.internal.OS
+
+import scala.collection.Factory
+import scala.jdk.StreamConverters._
+import sys.process._
+
+object FileSystem {
+  def listDirectory(dir: Path): Seq[Path] =
+    if (!Files.exists(dir)) Seq()
+    else Files.list(dir).toScala(Factory.arrayFactory).toSeq
+
+  def writeTextFile(path: Path, content: String): Unit = {
+    val writer = new PrintWriter(path.toFile)
+    try {
+      writer.write(content)
+    } finally {
+      writer.close()
+    }
+  }
+
+  def copyDirectory(source: Path, destination: Path): Unit =
+    FileUtils.copyDirectory(source.toFile, destination.toFile)
+
+  def removeDirectory(dir: Path): Unit =
+    FileUtils.deleteDirectory(dir.toFile)
+
+  def copyFile(source: Path, destination: Path): Unit =
+    FileUtils.copyFile(source.toFile, destination.toFile)
+
+  def ensureIsExecutable(file: Path): Unit = {
+    if (!Files.isExecutable(file)) {
+      def tryChmod(): Boolean = {
+        Seq("chmod", "+x", file.toAbsolutePath.toString).! == 0
+      }
+
+      if (OS.isWindows || !tryChmod()) {
+        throw new IllegalStateException(
```

Comment:
This looks like a very heavy-handed approach. Surely you could just print a warning and move on?

Reply:
Hmm, well I guess that would work, as long as they notice the warning.